docs: Updates Index.md file to show how to use `RpcException` property of `SpannerException`. (#9770)
googleapis/google-cloud-dotnet: apis/Google.Cloud.Spanner.Data/docs/index.md

the statistics for the session pool associated with the connection:

{{sample:SpannerConnection.GetSessionPoolSegmentStatistics}}

> ## RpcException from SpannerException

Reviewer: I think this heading needs to have a verb, e.g. "Obtaining an RpcException from a SpannerException" (or possibly Retrieving instead of Obtaining).

The `RpcException` property of `SpannerException` returns the underlying
`RpcException` which caused the `SpannerException`. The returned 
value may be null if the `SpannerException` was not caused by an `RpcException`.

> `ErrorInfo` in [RpcException](../../../docs/reference/help/troubleshooting#how-can-i-get-errorinfo-from-rpcexception) can 

Reviewer 2: This doesn't appear to be the right URL and I guess this link will be broken.
I am not sure of the convention we follow in docs and if url must be a relative url or the absolute web url. Please check with Jon. I do see few absolute urls as well. 
But if we are to use relative url - it should have been (../../../docs/devsite-help/troubleshooting.md#how-can-i-get-errorinfo-from-rpcexception) or the absolute url (if its allowed) should be [RpcException](https://cloud.google.com/dotnet/docs/reference/help/troubleshooting#how-can-i-get-errorinfo-from-rpcexception)

Author: Thanks Rishabh, updated the relative url as per your suggestion. @jskeet Please let us know that the relative url seems fine to you or we should use an absolute url pointing to RpcExceptions help page?

> `ErrorInfo` in [RpcException](../../../docs/devsite-help/troubleshooting.md#how-can-i-get-errorinfo-from-rpcexception) can 
> provide actionable error deatils. 

Reviewer: ```suggestion
This can in turn be used to [obtain an ErrorInfo](../../../docs/devsite-help/troubleshooting.md#how-can-i-get-errorinfo-from-rpcexception) which can provide actionable error details. 
```

> `RpcException` property of `SpannerException` can be used like:

Reviewer: ```suggestion
Sample code:
```

```csharp
using Google.Api.Gax.Grpc;
try
{
    // Code using Spanner. 
}
catch (SpannerException ex)
{

>     var rpcException = ex.RpcException;
>     if (rpcException != null)

Reviewer: ```suggestion
    if (ex.RpcException is RpcException rpcException)
```

    {

>         // Use the domain, reason and metadata of rpcException's ErrorInfo, e.g. for logging or taking remedial action.

Reviewer: Hmm... there are a couple of options here. We could show three scenarios:

- RpcException with ErrorInfo (including how to get it)
- RpcException without ErrorInfo
- No RpcException

Or we could just show RpcException with ErrorInfo, and no RpcException.

Author: Updated sample for all three scenarios.

the statistics for the session pool associated with the connection:

{{sample:SpannerConnection.GetSessionPoolSegmentStatistics}}

## Retrieving an RpcException from a SpannerException

The `RpcException` property of `SpannerException` returns the underlying
`RpcException` which caused the `SpannerException`. The returned 
value may be null if the `SpannerException` was not caused by an `RpcException`.

This can in turn be used to [obtain an ErrorInfo](../../../docs/devsite-help/troubleshooting.md#how-can-i-get-errorinfo-from-rpcexception) 

> which can provide actionable error details.. 

Reviewer: `..` => `.` at the end of the sentence

Author: Rectified.
